lib/cache/web_session.go

```diff
@@ -312,3 +318,101 @@ func (c *Cache) GetSnowflakeSession(ctx context.Context, req types.GetSnowflakeS
 	}
 	return out, trace.Wrap(err)
 }
+
+// RangeSnowflakeSessions returns Snowflake session resources within the range [start, end).
+func (c *Cache) RangeSnowflakeSessions(ctx context.Context, start, end string) iter.Seq2[types.WebSession, error] {
+	return func(yield func(types.WebSession, error) bool) {
+		ctx, span := c.Tracer.Start(ctx, "cache/RangeSnowflakeSessions")
+		defer span.End()
+
+		rg, err := acquireReadGuard(c, c.collections.snowflakeSessions)
+		if err != nil {
+			yield(nil, err)
+			return
+		}
+		defer rg.Release()
+
+		if rg.ReadCache() {
+			for a := range rg.store.resources(snowflakeSessionNameIndex, start, end) {
+				if !yield(a.Copy(), nil) {
+					return
+				}
+			}
```

Comment:
Are we ok with holding the cache read lock for the whole stream? IIRC for some other stream constructions in the cache we decided to collect small chunks at a time so the cache lock doesn't have to be held while yielding control to the caller's code.